engine/src/engine.ts

```diff
@@ -361,26 +363,23 @@ export default class Engine {
   addPlayer(player: Player) {
     this.players.push(player);
 
-    player.data.on(`gain-${Resource.TechTile}`, () => this.processNextMove(SubPhase.ChooseTechTile));
-    player.data.on(`gain-${Resource.TemporaryStep}`, () => this.processNextMove(SubPhase.BuildMine));
+    player.data.on(`gain-${Resource.TechTile}`, () => this.processNextMove(SubPhase.ChooseTechTile, null, true));
+    player.data.on(`gain-${Resource.TemporaryStep}`, () => this.processNextMove(SubPhase.BuildMine, null, true));
     player.data.on(`gain-${Resource.TemporaryRange}`, (count: number) => {
-      this.processNextMove(SubPhase.BuildMineOrGaiaFormer);
+      this.processNextMove(SubPhase.BuildMineOrGaiaFormer, null, true);
     });
-    player.data.on(`gain-${Resource.RescoreFederation}`, () => this.processNextMove(SubPhase.RescoreFederationTile));
-    player.data.on(`gain-${Resource.PISwap}`, () => this.processNextMove(SubPhase.PISwap));
-    player.data.on(`gain-${Resource.SpaceStation}`, () => this.processNextMove(SubPhase.SpaceStation));
+    player.data.on(`gain-${Resource.RescoreFederation}`, () =>
+      this.processNextMove(SubPhase.RescoreFederationTile, null, true)
+    );
+    player.data.on(`gain-${Resource.PISwap}`, () => this.processNextMove(SubPhase.PISwap, null, true));
+    player.data.on(`gain-${Resource.SpaceStation}`, () => this.processNextMove(SubPhase.SpaceStation, null, true));
     player.data.on(`gain-${Resource.DowngradeLab}`, () => {
-      this.processNextMove(SubPhase.DowngradeLab);
-      this.processNextMove(SubPhase.UpgradeResearch);
+      this.processNextMove(SubPhase.DowngradeLab, null, true);
+      this.processNextMove(SubPhase.UpgradeResearch, null, false);
```

Comment:
no need to specify arg 2/3

Reply:
wanted to make it explicit that research is not required.